internal/dag/dag.go

```diff
@@ -79,7 +80,14 @@ type HeaderCondition struct {
 }
 
 func (hc *HeaderCondition) String() string {
-	return "header: " + hc.Name
+	details := strings.Join([]string{
```

Comment:
Maybe there are cases where different routes are required. So maybe this is the way to go.

However an incoming request (method:GET, Prefix:/) will match both -

```
(HeaderName: ":method", MatchType: "Exact", HeaderValue: "GET")
(HeaderName: ":method", MatchType: "Contains", HeaderValue: "GET")

```
Allowing routes differing on MatchType **may** create duplicates.

Reply:
I think ideally we could detect if routes created won't match and set the status on the route, just not sure how difficult determining all those scenarios are. We'd have to match on duplicate header types of different types as @chintan8saaras noted.

Reply:
Seems like there are a couple of related improvements we could make here:

1. within a single route, determine when one condition is a proper subset of another (e.g. "exact: foo" and "contains: foo" on the same header), and collapse them into just a single condition (in this case, just "exact: foo").
2. when two routes can be combined similarly to the above, collapse them into a single route
3. when there are conflicting header conditions for a given route (e.g. both "contains: foo" and "notcontains: foo"), mark the proxy as invalid. This is already done for conflicting "exact" conditions, but we could strengthen this validation.